src/deprecated.jl

```diff
@@ -1,5 +1,12 @@
-import Base: @deprecate_binding, @deprecate
 
-# BEGIN Mocking 0.5 deprecations
+macro mock(expr)
+    w = Expr(:macrocall, Symbol("@warn"), __source__, "@mock is no longer required.")
```

Comment:
```julia
macro mock(expr)
    Base.depwarn("@mock is no longer required.", Symbol("@mock"))
    expr
end
```

Reply:
This get's the line numbers wrong.
Which makes cleaning out the extra deprecated `@mock` really annoying.

Reply:
Does this get the line numbers wrong?
```julia
macro mock(expr)
    quote
        Base.depwarn("`@mock` is no longer required.", Symbol("@mock"))
        $(esc(expr))
    end
end
```

Reply:
Yes,
```
julia> function foo()
       @mock identity(1)
       end
┌ Warning: `@mock` is no longer required.
│   caller = ip:0x0
└ @ Core :-1
```